src/disassembling/rzil.md

```diff
@@ -0,0 +1,178 @@
+[0x10000488]> aoi?
+Usage: aoi[p]   # Print the RzIL of next N instructions
+| aoi [<n_instructions>]  # Print the RzIL of next N instructions
+| aoip [<n_instructions>] # Pretty print the RzIL of next N instructions
+[0x10000488]> aoip 2
+0x10000488
+(seq
+  (storew 0
+    (+
+      (var r1)
+      (let v
+        (bv 16 0xfff0)
+        (ite
+          (msb
+            (var v))
+          (cast 32
+            (msb
+              (var v))
+            (var v))
+          (cast 32
+            false
+            (var v)))))
+    (cast 32
+      false
+      (var r1)))
+  (set r1
+    (+
+      (var r1)
+      (let v
+        (bv 16 0xfff0)
+        (ite
+          (msb
+            (var v))
+          (cast 32
+            (msb
+              (var v))
+            (var v))
+          (cast 32
+            false
+            (var v)))))))
+0x1000048c
+(set r0
+  (cast 32
+    false
+    (var lr)))
+```
+
+The `plf` command allows to generate an in-line representation of the entire function in s-expressions.
+
+```bash
+[0x100002bc]> pdf @ sym.example
+/ sym.example();
+|           ; var int32_t var_1ch @ stack - 0x1c
+|           0x1000044c      stwu  r1, -0x10(r1)
+|           0x10000450      mflr  r0
+|           0x10000454      stw   r0, 0x14(r1)
+|           0x10000458      lwz   r0, 0x14(r1)
+|           0x1000045c      addi  r1, r1, 0x10
+|           0x10000460      mtlr  r0
+\           0x10000464      blr
+[0x100002bc]> plf @ sym.example
+0x1000044c (seq (storew 0 (+ (var r1) (let v (bv 16 0xfff0) (ite (msb (var v)) (cast 32 (msb (var v)) (var v)) (cast 32 false (var v))))) (cast 32 false (var r1))) (set r1 (+ (var r1) (let v (bv 16 0xfff0) (ite (msb (var v)) (cast 32 (msb (var v)) (var v)) (cast 32 false (var v)))))))
+0x10000450 (set r0 (cast 32 false (var lr)))
+0x10000454 (seq (storew 0 (+ (var r1) (let v (bv 16 0x14) (ite (msb (var v)) (cast 32 (msb (var v)) (var v)) (cast 32 false (var v))))) (cast 32 false (var r0))) empty)
+0x10000458 (seq (set r0 (let ea (+ (var r1) (let v (bv 16 0x14) (ite (msb (var v)) (cast 32 (msb (var v)) (var v)) (cast 32 false (var v))))) (let loadw (loadw 0 32 (var ea)) (cast 32 false (var loadw))))) empty)
+0x1000045c (seq (set a (var r1)) (set b (let v (bv 16 0x10) (ite (msb (var v)) (cast 32 (msb (var v)) (var v)) (cast 32 false (var v))))) empty (set r1 (+ (var a) (var b))) empty empty empty)
+0x10000460 (set lr (cast 32 false (var r0)))
+0x10000464 (seq (set CIA (bv 32 0x10000464)) empty empty (set NIA (& (bv 32 0xfffffffc) (var lr))) (jmp (var NIA)))
+[0x100002bc]> 
+```
+
+The same output of `aoi` can be obtained via `rz-asm` like this:
+
+```bash
+$ rz-asm -de -a ppc 7c0802a6
+mflr r0
+$ rz-asm -Ie -a ppc 7c0802a6
+(set r0 (cast 32 false (var lr)))
+```
+
+## Emulation
+
+Rizin enables instruction emulation by leveraging RzIL. The emulation can be used to record changes within the VM, like read and writes of registers and memory locations. The emulation is controlled via the `aez` commands.
```

Comment:
it requires `e io.buffers=true` currently. Otherwise memory read/writes don't work. Should we mention this here?
It took me like two times +20min to figure this out.

Reply:
yes, we should mention this.